### ISSUE TITLE
console.log(object) not supported?  Code optimized out?

### ISSUE THREAD

**Issue by NoCheroot** [NectarJS/nectarjs#13]
Compiled the following:

testObject.exe
```
var testObject = { testKey1: "test 1" };
testObject["testKey2"] = "test 2";
console.log(testObject);
```
testObject2.exe
```
console.log("test 0");
var testObject = { testKey1: "test 1" };
testObject["testKey2"] = "test 2";
console.log(testObject);
console.log("test 3");
```
testObjectKey.exe
```
var testObject = { testKey1: "test 1" };
testObject["testKey2"] = "test 2";
console.log(testObject["testKey1"]);
```

Their outputs look like this:
![2018-01-03 09_00_39-win 10 dt on lt-01012252 - virtual machine connection](https://user-images.githubusercontent.com/10468415/34525843-eba80f36-f065-11e7-95b2-13e394c955d2.png)

First, look at **testObject2.exe**.  Strings "test 0" and "test 3" made it into the binary, but the others did not.
![2018-01-03 08_28_32- cmd exe - radare2 _result-win-x86-64 testobject2 exe_ - gvim](https://user-images.githubusercontent.com/10468415/34525967-72266ce2-f066-11e7-8609-1fbf1817d90a.png)

"test 0" and "test 3" are puts() by this, but code related to the object testObject is missing:
![2018-01-03 08_31_23- cmd exe - radare2 _result-win-x86-64 testobject2 exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526031-b062ed3c-f066-11e7-8382-35778a3c010d.png)

sub.test_0_540 is called here, mixed in with initialization code and exit which all NectarJS win-x86-64 targets seem to share (boilerplate?  template?):
![2018-01-03 08_33_13- cmd exe - radare2 _result-win-x86-64 testobject2 exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526131-173f202a-f067-11e7-980d-69d45430d6f0.png)

Now **testObjectKey.exe**.  All related strings are present in the binary.
![2018-01-03 08_56_55- cmd exe - radare2 _result-win-x86-64 testobjectkey exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526150-2bafbfe2-f067-11e7-84c4-b2b298b54fcd.png)

Looks like code working with the object is here:
![2018-01-03 08_57_14- cmd exe - radare2 _result-win-x86-64 testobjectkey exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526257-948b2df8-f067-11e7-815c-b95adc028ae0.png)

Which is called here, with familiar boilerplate:
![2018-01-03 08_57_55- cmd exe - radare2 _result-win-x86-64 testobjectkey exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526536-57fe80fa-f068-11e7-9e00-38d392870d76.png)

Lastly, **testObject.exe**.  All of the related strings are missing.
![2018-01-03 08_34_30- cmd exe - radare2 _result-win-x86-64 testobject exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526453-1e50b652-f068-11e7-9c6f-a1d556175c14.png)

In the spot where I should find a call to the code which works with the object, I instead find a call to some code which has another call to fcn.00401f00.  I stopped here and have not put any effort into understanding what fcn.00401f00 does since it, or an equivalent, appears to be called during initialization in any case (the cases I've looked at, anyway).
![2018-01-03 08_38_16- cmd exe - radare2 _result-win-x86-64 testobject exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526772-1b264e78-f069-11e7-9270-864ea2f321e5.png)
![2018-01-03 08_45_36- cmd exe - radare2 _result-win-x86-64 testobject exe_ - gvim](https://user-images.githubusercontent.com/10468415/34526778-1f5119ce-f069-11e7-95f9-c5027a18838f.png)

Basically, it looks like I ended up with an exe with nothing but boilerplate.

Is console.log(object) not yet supported?  Or, did something go wrong during the build, the problem was handled gracefully, and the exe was generated anyway?  Maybe it would be better for the build to fail.

Assuming the object and its supporting strings and code were optimized out because there are no useful references to it, where did the optimization occur?  Your compiler, or by MinGW?

**Comment by adrien-thierry** [comment]
both (console && objects) are in work, I let this open to add news about it

But you're right, it should (and will) failed if it's not implemented
